protocol-units/sequencing/memseq/src/lib.rs

```diff
@@ -404,65 +413,78 @@ pub mod test {
 	}
 
 	/// Mock Mempool
+
+	// create a macro generating an Err(MempoolTransactionOperationsError::MockError("message")) out
+	// of "message"
+
+	macro_rules! mock_error {
+		($message:expr) => {
+			Err(mempool_util::MempoolTransactionOperationsError::MockError($message.to_string()))
+		};
+	}
+
 	struct MockMempool;
 	impl MempoolTransactionOperations for MockMempool {
 		async fn has_mempool_transaction(
 			&self,
 			_transaction_id: Id,
-		) -> Result<bool, anyhow::Error> {
-			Err(anyhow::anyhow!("Mock has_mempool_transaction"))
+		) -> MempoolTransactionOperationsResult<bool> {
+			mock_error!("has_mempool_transaction")
```

Comment:
This should just be left at `todo!` until there is a meaningful mocked implementation that is capable of simulating real success and error cases.

Reply:
@mzabaluev why should this just be left at `todo!`? We prefer to `panic!` instead of returning an Err

Reply:
If the mock cannot provide a useful implementation to support the tests at this point, it's better to panic up front, which is what `todo!` does. When we have code under test that actually calls the method, we'll naturally need to supply something life-like, rather than a "mock error", from the mock side.